parcels/particleset.py

```diff
@@ -161,7 +161,7 @@ def ArrayClass_init(self, *args, **kwargs):
             raise NotImplementedError("If fieldset.time_origin is not a date, time of a particle must be a double")
         time = np.array([self.time_origin.reltime(t) if _convert_to_reltime(t) else t for t in time])
         assert lon.size == time.size, "time and positions (lon, lat, depth) do not have the same lengths."
-        if isinstance(fieldset.U, Field) and (not fieldset.U.allow_time_extrapolation):
+        if isinstance(fieldset.U, Field) and (len(fieldset.U.data.time) > 1):
```

Comment:
This would just be:

```suggestion
        if fieldset.time_interval:
```

And then we can update `_warn_particle_times_outside_fieldset_time_bounds` to take `release_times: np.array, time_interval: TimeInterval`

Reply:
Changes made in https://github.com/OceanParcels/Parcels/pull/1999/commits/2afafa0740efdc42e19ac66575b898f1a60ac721